pkg/auth/user/RoleGroupService.go

```diff
@@ -619,12 +624,67 @@ func (impl RoleGroupServiceImpl) FetchDetailedRoleGroups() ([]*bean.RoleGroup, e
 	return list, nil
 }
 
-func (impl RoleGroupServiceImpl) FetchRoleGroups() ([]*bean.RoleGroup, error) {
-	roleGroup, err := impl.roleGroupRepository.GetAllRoleGroup()
+func (impl RoleGroupServiceImpl) FetchRoleGroups() (*bean.RoleGroupListingResponse, error) {
+	list, err := impl.FetchDetailedRoleGroups()
 	if err != nil {
-		impl.logger.Errorw("error while fetching user from db", "error", err)
+		impl.logger.Errorw("error in FetchDetailedRoleGroups", "err", err)
+		return nil, err
+	}
+	response := &bean.RoleGroupListingResponse{
+		RoleGroups: list,
+		TotalCount: len(list),
+	}
+	return response, nil
+}
+
+// FetchRoleGroupsWithFilters takes filters args as input and outputs RoleGroupListingResponse based on the request filters.
+func (impl RoleGroupServiceImpl) FetchRoleGroupsWithFilters(sortOrder string, sortBy string, offset int, totalSize int, showAll bool, searchKey string) (*bean.RoleGroupListingResponse, error) {
+	// default values will be used if not provided
+	request := impl.getRequestWithFiltersArgs(sortOrder, sortBy, offset, totalSize, showAll, searchKey)
+	if request.ShowAll {
+		return impl.FetchRoleGroups()
+	}
+	// Setting size as zero to calculate the total number of results based on request
+	size := request.Size
+	request.Size = 0
+
+	query := impl.userRepositoryQueryBuilder.GetQueryForGroupListingWithFilters(request)
+	totalCount, err := impl.userRepository.GetCountExecutingQuery(query)
+	if err != nil {
+		impl.logger.Errorw("error in FetchRoleGroupsWithFilters", "err", err, "query", query)
+		return nil, err
+	}
+
+	request.Size = size
+	query = impl.userRepositoryQueryBuilder.GetQueryForGroupListingWithFilters(request)
+	roleGroup, err := impl.roleGroupRepository.GetAllExecutingQuery(query)
+	if err != nil {
+		impl.logger.Errorw("error while FetchRoleGroupsWithFilters", "error", err, "query", query)
 		return nil, err
 	}
+
+	list := impl.fetchRoleGroupResponseFromModel(roleGroup)
+	response := &bean.RoleGroupListingResponse{
+		RoleGroups: list,
+		TotalCount: totalCount,
+	}
+	return response, nil
+}
+
+func (impl RoleGroupServiceImpl) getRequestWithFiltersArgs(sortOrder string, sortBy string, offset int, totalSize int, showAll bool, searchKey string) *bean.FetchListingRequest {
+	sortByRes, size := impl.userCommonService.GetDefaultValuesIfNotPresent(sortBy, totalSize, true)
```

Comment:
pass object, get object with updated values makes it easy to introduce more defaults when needed

Reply:
DONE